interfaces/mount/spec.go

```diff
@@ -206,3 +216,31 @@ func (spec *Specification) AddPermanentSlot(iface interfaces.Interface, slot *sn
 	}
 	return nil
 }
+
+// AddOvername records mappings of snap directories.
+//
+// When the snap is installed with an instance key, set up it's mount namespace
+// such that it appears as a non-instance key snap. This ensures compatibility
+// with code making assumptions about $SNAP{,_DATA,_COMMON} locations. That is,
+// given a snap foo_bar, the mappings added are:
+//
+// - /snap/foo_bar      -> /snap/foo
+// - /var/snap/foo_bar  -> /var/snap/foo
+func (spec *Specification) AddOvername(info *snap.Info) {
+	if info.InstanceKey == "" {
+		return
+	}
+
+	// /snap/foo_bar -> /snap/foo
+	spec.AddOvernameMountEntry(osutil.MountEntry{
+		Name:    path.Join(dirs.CoreSnapMountDir, info.InstanceName()),
+		Dir:     path.Join(dirs.CoreSnapMountDir, info.SnapName()),
+		Options: []string{"rbind", osutil.XSnapdOriginOvername()},
+	})
+	// /var/snap/foo_bar -> /var/snap/foo
+	spec.AddOvernameMountEntry(osutil.MountEntry{
+		Name:    path.Join(dirs.SnapDataDir, info.InstanceName()),
+		Dir:     path.Join(dirs.SnapDataDir, info.SnapName()),
+		Options: []string{"rbind", osutil.XSnapdOriginOvername()},
+	})
```

Comment:
Should we add the overname for `/home/snap/${SNAP_NAME}_${SNAP_INSTANCE_KEY}` -> `/home/snap/$SNAP_NAME/`?

Reply:
This was discussed with @niemeyer and we have decided to drop the user mounts at this point. We can do a quick meeting make sure that we're all on the same page. WDYT?

Reply:
Let's at least postpone it so we can get this PR through and the feature working.